### DISCUSSION TITLE
Is HTML Report generation with screenshots available in flutter like extent reports.?

### DISCUSSION THREAD

**Discussion by lravihs** [leancodepl/patrol#1526] [Q&A]
### Steps to reproduce

Have to generate HTML reports like extent reports with attached screenshots

### Actual results

Have to generate HTML reports like extent reports with attached screenshots

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. Do not remove the backticks! -->
```

</details>

### Patrol version

patrol:^2.0.3

### Patrol Doctor output

patrol doctor

### Flutter Doctor output

flutter doctor

**Comment by bartekpacia** [comment]
Yes, it's possible with the native JUnit XML / `xcresult` report that's generated after `patrol test`. There are many tools that can generate a nice HTML from those results, e.g. https://github.com/kishikawakatsumi/xcresulttool.


On Android, the results are in ` build/app/outputs/androidTest-results/`, on iOS, the path is printed after `patrol test`:

```
...
Test Case '-[RunnerUITests exampleTest]' passed (7.183 seconds).
	Test Suite 'RunnerUITests' passed at 2023-07-13 12:05:11.885.
		 Executed 1 test, with 0 failures (0 unexpected) in 7.183 (7.185) seconds
	Test Suite 'RunnerUITests.xctest' passed at 2023-07-13 12:05:11.887.
		 Executed 1 test, with 0 failures (0 unexpected) in 7.183 (7.187) seconds
	Test Suite 'All tests' passed at 2023-07-13 12:05:11.888.
		 Executed 1 test, with 0 failures (0 unexpected) in 7.183 (7.190) seconds
	2023-07-13 12:05:12.176 xcodebuild[25363:3442235] [MT] IDETestOperationsObserverDebug: 16.322 elapsed -- Testing started completed.
	2023-07-13 12:05:12.176 xcodebuild[25363:3442235] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
	2023-07-13 12:05:12.176 xcodebuild[25363:3442235] [MT] IDETestOperationsObserverDebug: 16.322 sec, +16.322 sec -- end
	2023-07-13 12:05:12.265 xcodebuild[25363:3442235] [MT] IDEResultKit: Warning: While writing a result bundle to /Users/bartek/Library/Developer/Xcode/DerivedData/Runner-aqensjxfrjjuawbcawtvgrjvriuk/Logs/Test/Test-Runner-2023.07.13_12-04-55-+0200.xcresult, not all contents have been imported, as the Staging directory still contains the following files/directories: ["1_Test"]

	Test session results, code coverage, and logs:
		/Users/bartek/Library/Developer/Xcode/DerivedData/Runner-aqensjxfrjjuawbcawtvgrjvriuk/Logs/Test/Test-Runner-2023.07.13_12-04-55-+0200.xcresult

	** TEST EXECUTE SUCCEEDED **
...
```

**Comment by pandoras-toolbox** [comment]
Maybe some time Allure (see a [demo report](https://allure-framework.github.io/allure-demo/7/)) can be used for Dart/Flutter/Patrol tests to have a really good report. I had asked the question here about Allure, but to implement it the right way takes some effort, and I think its currently not such a high priority if the effort is high (which it is I believe).

**Comment by lravihs** [comment]
@pandoras-toolbox  I have checked the ALLURE REPORT official website and they didn't mention the support for dart/flutter. Can you post if you have some samples for the allure report for DART/FLUTTER.
NOTE : Kindly refer the following link

https://docs.qameta.io/allure/#_gradle_6

**Comment by pandoras-toolbox** [comment]
@lravihs [Here](https://github.com/leancodepl/patrol/discussions/1471) is the discussion about Allure/Patrol. There I mention that Allure currently does not support Dart unfortunately. The Allure owners do not use Flutter so they rely on someone contributing it.

**Reply by bartekpacia** [reply]
From what I understand, Allure doesn't have to explicitly support Dart. It's enough that we have JUnit reports and we feed those reports into Allure.

**Reply by lravihs** [reply]
That's Awesome..!

If it is so then We need to find a way to feed JUnit reports into Allure. Please provide the link here if someone find the way to use it

**Reply by pandoras-toolbox** [reply]
@lravihs That will only beautify the test report but not use the many useful features from Allure at all. And that kind of Allure report creation cannot be improved later to add these features, because the support for Dart would have to be done in another way. See the [answer](https://github.com/allure-framework/allure2/issues/1119#issuecomment-1635981998) to my question in the Allure forum from baev. If that is good enough for you, fine, just to let you know.
